drv/sbrmi/src/main.rs

```diff
@@ -0,0 +1,156 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Task for communicating to the host CPU via the SB-RMI interface.
+
+#![no_std]
+#![no_main]
+
+use drv_i2c_devices::sbrmi::{CpuidResult, Sbrmi};
+use drv_sbrmi_api::SbrmiError;
+use idol_runtime::RequestError;
+use ringbuf::*;
+use userlib::*;
+use zerocopy::FromBytes;
+
+include!(concat!(env!("OUT_DIR"), "/i2c_config.rs"));
+
+struct ServerImpl;
+
+task_slot!(I2C, i2c_driver);
+
+#[derive(Copy, Clone, PartialEq)]
+enum Trace {
+    None,
+    CpuidError(drv_i2c_devices::sbrmi::Error),
+    CpuidResult(CpuidResult),
+    Rdmsr(u32),
+    RdmsrError(drv_i2c_devices::sbrmi::Error),
+    RdmsrOk,
+}
+
+ringbuf!(Trace, 16, Trace::None);
+
+fn sbrmi() -> Sbrmi {
+    let devs = i2c_config::devices::sbrmi(I2C.get_task_id());
+    Sbrmi::new(&devs[0])
```

Comment:
It would be a little more efficient to store this in the `ServerImpl`, so you don't need to rebuild it every time.

Reply:
Done